src/annotate.py

```diff
@@ -19,25 +40,25 @@ def annotate_img(image, boxes, image_size):
         confidence = box[1]
 
         # dimensions of box
-        x, y, w, h = box[2:]
+        xmin, xmax, ymin, ymax = box[2:]
         
-        # calculate vertices
-        x0 = max(y - w/2, 0)
-        x1 = min(y + w/2, image_size[0] - 1)
-        y0 = max(x - h/2, 0)
-        y1 = min(x + h/2, image_size[1] - 1)
+        # limit vertices to image
+        xmin = max(xmin, 0)
+        xmax = min(xmax, img_size[0] - 1)
+        ymin = max(ymin, 0)
+        ymax = min(ymax, img_size[1] - 1)
 
         # get color
         color = colors[labels.index(label)%len(colors)]
 
         # draw rectangle
-        draw.rectangle([x0, y0, x1, y1], outline=color, width=5)
+        draw.rectangle([xmin, ymin, xmax, ymax], outline=color, width=5)
 
         # draw text
-        txt_size = min(15, int(image_size[1]/40))
+        txt_size = min(15, int(img_size[1]/40))
         font = ImageFont.truetype('Pillow/Tests/fonts/DejaVuSans.ttf', txt_size)
-        y0 = max(y0, txt_size)
+        ymin = max(ymin, txt_size)
         text = label + '-' + '{:.1f}'.format(confidence*100) + '%'
```

Comment:
maybe you could use [python f-strings](https://realpython.com/python-f-strings/)?